source/common/network/io_socket_handle_impl.cc

```diff
@@ -576,5 +576,54 @@ absl::optional<std::chrono::milliseconds> IoSocketHandleImpl::lastRoundTripTime(
   return std::chrono::duration_cast<std::chrono::milliseconds>(info.tcpi_rtt);
 }
 
+absl::optional<std::string> IoSocketHandleImpl::interfaceName() {
+  auto& os_syscalls_singleton = Api::OsSysCallsSingleton::get();
+  if (!os_syscalls_singleton.supportsGetifaddrs()) {
+    return absl::nullopt;
+  }
+
+  Address::InstanceConstSharedPtr socket_address = localAddress();
+  if (!(socket_address && socket_address->type() == Address::Type::Ip)) {
```

Comment:
nit: I would suggest that we expand this into an `||` because I think it's simpler to reason about without the compound negation.